File: cpp/1528.shuffle-string.cpp
class Solution {
public:
    string solve(string &s, vector<int> &indices){
	   string res(s.size(), ' ');
	   for(int i = 0 ; i < s.size(); i++){
	      res[indices[i]] = s[i];
	   }
	   return res;
    }
    string restoreString(string s, vector<int>& indices) {
	    return solve(s, indices);
    }
};

File: python/1528.shuffle-string.py
class Solution:
    def solve(self, s, indices):
        res = ['']*len(s)
        for i in range(len(s)):
            res[indices[i]] = s[i]
        return ''.join(res)

    def restoreString(self, s: str, indices: List[int]) -> str:
        return self.solve(s, indices)
        
